File: app/src/main/java/net/tlalka/android/fiszki/core/components/ApplicationComponent.java
package net.tlalka.android.fiszki.core.components;

import javax.inject.Singleton;

import dagger.Component;
import net.tlalka.android.fiszki.core.modules.ApplicationModule;
import net.tlalka.android.fiszki.core.modules.SessionModule;

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    SessionComponent add(SessionModule sessionModule);
}

File: app/src/main/java/net/tlalka/android/fiszki/core/components/ApplicationComponent.kt
package net.tlalka.android.fiszki.core.components

import dagger.Component
import net.tlalka.android.fiszki.core.modules.ApplicationModule
import net.tlalka.android.fiszki.core.modules.SessionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationModule::class])
interface ApplicationComponent {

    fun add(sessionModule: SessionModule): SessionComponent
}
